### PR TITLE
feat(app and service): update subscription logic

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**
## Description

App and Service subscription logic update

## Why

As per backend API response there were logic updates in FE

## Issue

n/a

## Checklist

Please delete options that are not relevant.

- [X] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [X] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [X] I have performed a self-review of my own code
- [X] I have successfully tested my changes locally

### CODE REVIEW

**Comment by evegufy** [comment]
@nidhigarg-bmw thanks for the PR!
could you please update the dependencies file?

**Comment by nidhigarg-bmw** [comment]
> @nidhigarg-bmw thanks for the PR! could you please update the dependencies file?

Just did

File: src/features/appSubscription/appSubscriptionApiSlice.ts [hunk 1]
```diff
@@ -84,12 +88,12 @@ export type CompanySubscriptionData = {
   subscriptionId: string
   offerSubscriptionStatus: string
   technicalUser?: boolean
+  processStepTypeId: ProcessStep
```
Comment (Phil91) [inline src/features/appSubscription/appSubscriptionApiSlice.ts:91]: the processStepTypeId can be null. That's the case if the process is done processStepTypeId `null` will be returned.

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/features/appSubscription/appSubscriptionApiSlice.ts (hunk 1 above)
Comment (nidhigarg-bmw) [reply src/features/appSubscription/appSubscriptionApiSlice.ts:91]: Yes, I checked on API response and already added conditions for null case. But again I added the null type here as well

**Review by manojava-gk — APPROVED** [review]
lftm

**Review by lavanya-bmw — APPROVED** [review]
lgtm

**Review by Phil91 — APPROVED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**